app/components/CrashDiagramCard.tsx

```diff
@@ -2,63 +2,162 @@ import { useState } from "react";
+        onChange={rotate}
+        style={{ width: "70%" }}
+        title="Rotate Diagram"
+      />
+    </div>
+  );
+};
+
 export default function CrashDiagramCard({ crash }: { crash: Crash }) {
   const [diagramError, setDiagramError] = useState(false);
+  const [rotation, setRotation] = useState(0);
 
   return (
     <Card>
       <Card.Header>Diagram</Card.Header>
       <Card.Body className="crash-header-card-body text-center">
-        {!diagramError && (
-          <Image
-            fluid
-            style={{
-              height: "100%",
-              maxWidth: "100%",
-            }}
-            src={`${CR3_DIAGRAM_BASE_URL}/${crash.record_locator}.jpeg`}
-            alt="crash diagram"
-            onError={() => {
-              console.error("Error loading CR3 diagram image");
-              setDiagramError(true);
-            }}
-          />
-        )}
-        {diagramError && crash.is_temp_record && (
-          <Alert variant="info">
-            <i className="fa fa-info-circle" /> Crash diagrams are not available
-            for user-created crash records
-          </Alert>
-        )}
-        {diagramError && !crash.is_temp_record && (
-          <Alert variant="danger">
-            <p>
-              The crash diagram is not available. Typically, this indicates
-              there was an error when processing this crash&aposs CR3 PDF.
-            </p>
-            <p>
-              For additional assistance, you can&nbsp;
-              <a
-                href="https://atd.knack.com/dts#new-service-request/?view_249_vars=%7B%22field_398%22%3A%22Bug%20Report%20%E2%80%94%20Something%20is%20not%20working%22%2C%22field_399%22%3A%22Vision%20Zero%20Editor%22%7D"
-                target="_blank"
-                rel="noopener noreferrer"
-              >
-                report a bug
-                <i className="fa fa-external-link"></i>
-              </a>
-              .
-            </p>
-          </Alert>
-        )}
+        <TransformWrapper initialScale={1}>
+          {!diagramError && <ZoomResetControls setRotation={setRotation} />}
+          <TransformComponent>
+            <div style={{ height: "340px", width: "100%", overflow: "hidden" }}>
+              {!diagramError && (
+                <Image
+                  fluid
+                  style={{
+                    height: "100%",
+                    width: "440px",
+                    transform: `rotate(${rotation}deg)`,
+                    objectFit: "contain",
+                  }}
+                  src={`${CR3_DIAGRAM_BASE_URL}/${crash.record_locator}.jpeg`}
+                  alt="crash diagram"
+                  onError={() => {
+                    console.error("Error loading CR3 diagram image");
+                    setDiagramError(true);
+                  }}
+                />
+              )}
+              {diagramError && crash.is_temp_record && (
+                <Alert variant="info" style={{ marginTop: "50px" }}>
```

Comment:
nice—i agree that moving this alert toward the vertical center of the card is a better look, but i would prefer to use flexbox utilities to center this vertically. we have a few uses of `align-items-center` if you want to check that out 👀

Reply:
Thanks @johnclary! 

Can you clarify -- you noted that you prefer it scooted up to the top as a better look, but ask for `align-items-center`. The scooted-up version would be using `align-items-start` i think, and if i use `align-items-center`, it's centered at the midpoint of the vertical space. 

Also, I don't think bootstrap has a utility that let's me define a custom width of 490px, so I can center properly. Any ideas?

PS: This is like my tRPC comment -- I fully dig how we're doing it, but here's a thing I loved about tailwind. They have a solution for when you really need a hardcoded value, like here, you'd be able to use the class `w-[490px]` and it would end up making the custom width utility for you out of that.